tests/gpflow/posteriors/test_posteriors.py

```diff
@@ -0,0 +1,445 @@
+
+
+@pytest.fixture(name="whiten", params=[False, True])
+def _whiten_fixture(request):
+    return request.param
+
+
+@pytest.fixture(name="precompute", params=[False, True])
+def _precompute_fixture(request):
+    return request.param
+
+
+_independent_single_output = [IndependentPosteriorSingleOutput]
+_fully_correlated_multi_output = [FullyCorrelatedPosterior]
+_independent_multi_output = [IndependentPosteriorMultiOutput]
+_fallback_independent_multi_output = [FallbackIndependentLatentPosterior]
+_linear_coregionalization = [LinearCoregionalizationPosterior]
+
+
+def test_no_missing_kernels():
+    tested_kernels = set(
+        chain(
+            _independent_single_output,
+            _fully_correlated_multi_output,
+            _independent_multi_output,
+            _fallback_independent_multi_output,
+            _linear_coregionalization,
+        )
+    )
+
+    available_kernels = list(gpflow.ci_utils.subclasses(AbstractPosterior))
+    concrete_kernels = set([k for k in available_kernels if not isabstract(k)])
+
+    assert tested_kernels == concrete_kernels
+
+
+@dataclass(frozen=True)
+class LatentVariationalParameters:
+    qsqrt = (np.random.randn(NUM_INDUCING_POINTS, 1) ** 2) * 0.01
+    qmean = np.random.randn(NUM_INDUCING_POINTS, 1)
+
+
+def _assert_fused_predict_f_equals_precomputed_predict_f(
+    posterior, precompute, full_cov, full_output_cov, decimals=6
+):
+    Xnew = np.random.randn(20, 1)
+
+    fused_f_mean, fused_f_cov = posterior.fused_predict_f(
+        Xnew, full_cov=full_cov, full_output_cov=full_output_cov
+    )
+
+    if not precompute:
+        posterior.update_cache()
+
+    precomputed_f_mean, precomputed_f_cov = posterior.predict_f(
+        Xnew, full_cov=full_cov, full_output_cov=full_output_cov
+    )
+
+    np.testing.assert_array_almost_equal(fused_f_mean, precomputed_f_mean, decimal=decimals)
+    np.testing.assert_array_almost_equal(fused_f_cov, precomputed_f_cov, decimal=decimals)
+
+
+@pytest.mark.parametrize("posterior_class", _independent_single_output)
+def test_independent_single_output(
+    posterior_class, whiten, mean_function, precompute, full_cov, full_output_cov
+):
+    kernel = gpflow.kernels.SquaredExponential()
+    inducing_variable = inducingpoint_wrapper(np.random.randn(NUM_INDUCING_POINTS, 1))
+
+    posterior = posterior_class(
+        kernel=kernel,
+        inducing_variable=inducing_variable,
+        q_mu=LatentVariationalParameters.qmean,
+        q_sqrt=tf.constant(LatentVariationalParameters.qsqrt),
+        whiten=whiten,
+        mean_function=mean_function,
+        precompute=precompute,
+    )
+
+    _assert_fused_predict_f_equals_precomputed_predict_f(
+        posterior, precompute, full_cov, full_output_cov
+    )
+
+
+@dataclass(frozen=True)
+class LatentVariationalMultiOutputParameters:
+    qsqrt = np.eye(NUM_INDUCING_POINTS)[None, :]
+    qmean = np.random.randn(NUM_INDUCING_POINTS, 1)
```

Comment:
would be really good to run the tests with more than 1 latent function

Reply:
Good point! I have added parameters for the number of input dimensions, the number of latent functions and the number of output dimensions.